File: src/main/java/kr/co/lunatalk/domain/order/repository/OrderRepository.java
package kr.co.lunatalk.domain.order.repository;

import kr.co.lunatalk.domain.order.domain.Order;
import kr.co.lunatalk.domain.order.domain.OrderStatus;
import org.springframework.data.jpa.repository.JpaRepository;

import java.time.LocalDateTime;

public interface OrderRepository extends JpaRepository<Order, Long>, OrderRepositoryCustom {
	long countByCreatedAtBetween(LocalDateTime start, LocalDateTime end);

	long countByStatusAndCreatedAtBetween(OrderStatus status, LocalDateTime start, LocalDateTime end);
}

File: lunatalk-core/src/main/kotlin/kr/co/lunatalk/domain/order/repository/OrderRepository.kt
package kr.co.lunatalk.domain.order.repository

import kr.co.lunatalk.domain.order.domain.Order
import kr.co.lunatalk.domain.order.domain.OrderStatus
import org.springframework.data.jpa.repository.JpaRepository
import java.time.LocalDateTime

interface OrderRepository : JpaRepository<Order, Long>, OrderRepositoryCustom {

    fun countByCreatedAtBetween(start: LocalDateTime, end: LocalDateTime): Long

    fun countByStatusAndCreatedAtBetween(status: OrderStatus, start: LocalDateTime, end: LocalDateTime): Long
}
